### PR TITLE
Handle no delta or output of compute task

### PR DESCRIPTION
**Pull request by lenak25**

### CODE REVIEW

**Comment by lenak25** [comment]
@AvishaiW , this PR is ready for your review..

File: test/ipc_test.js [hunk 1]
```diff
@@ -229,3 +229,41 @@ it('#6 GetPTTRequest *with* addresses - mock server', async function() {
         });
   });
 });
+
+it('#7 GetTips - mock server', async function() {
```
Comment (AvishaiW) [inline test/ipc_test.js]: would be great if you'll be able to explain to me what this test does exactly :)

File: src/worker/controller/actions/ethereum/CommitReceiptAction.js [hunk 2]
```diff
@@ -13,63 +13,121 @@ class CommitReceiptAction {
...
+    if (task.getResult().isSuccess()) {
+      res = await this._commitSuccessTask(task);
+    } else {
+      res = await this._commitFailedTask(task);
     }
-    throw new errors.TypeErr(`wrong type or missing fields in Result`);
+    return res;
   }
-  _commitFailedTask(task) {
+  async _commitFailedTask(task) {
```
Comment (AvishaiW) [inline src/worker/controller/actions/ethereum/CommitReceiptAction.js]: I saw you added `result` to many ipc messages, was wondering if the [failed task](https://github.com/enigmampc/enigma-p2p/blob/develop/docs/IPC_MESSAGES.md#failedtask-error-message) is also done accordingly, and if so would be great if you can point me to it :)

File: src/ethereum/EthereumVerifier.js [hunk 3]
```diff
@@ -284,49 +288,189 @@ class EthereumVerifier {
...
   }
 
+  _checkDeployResult(task, deltaHash, codeHash) {
+    let res = {isVerified: false, error: null};
+
+    if (!task.hasDelta()) {
+      res.error = new errors.TaskVerificationErr("No delta in task result " + task.getTaskId());
+    }
+    else if (!task.getOutput()) {
+      res.error = new errors.TaskVerificationErr("No output in task result " + task.getTaskId());
```
[the remark this answers is not in the document]
Comment (lenak25) [reply src/ethereum/EthereumVerifier.js]: Yes, the byte-code is its output

File: src/worker/controller/actions/tasks/VerifyAndStoreResultAction.js [hunk 4]
```diff
@@ -34,97 +34,119 @@ class VerifyAndStoreResultAction {
...
-    if (isVerified) {
-      const coreMsg = this._buildIpcMsg(resultObj, type, contractAddress);
-      if (coreMsg) {
-        try {
-          await this._controller.asyncExecCmd(constants.NODE_NOTIFICATIONS.UPDATE_DB, {data: coreMsg});
-        }
-        catch(e) {
-          if(optionalCallback){
-            return optionalCallback(e);
+    let {taskResult, err} = this._buildTaskResult(type, resultObj);
```
[the remark this answers is not in the document]
Comment (lenak25) [reply src/worker/controller/actions/tasks/VerifyAndStoreResultAction.js]: Task is too generic.. In the code we have a [Task object](https://github.com/enigmampc/enigma-p2p/blob/master/src/worker/tasks/Task.js) which holds the entire task information (its inputs and its result once it is done). 
Result is the output of the task which is being broadcasted to the network and each worker receiving it calls this action.

File: src/worker/controller/actions/ethereum/CommitReceiptAction.js (hunk 2 above)
Comment (lenak25) [reply src/worker/controller/actions/ethereum/CommitReceiptAction.js]: Yes, look [here](https://github.com/enigmampc/enigma-p2p/blob/master/src/worker/controller/actions/tasks/ExecuteVerifiedAction.js#L54)

**Pull request merged**